tikv/raftstore/peer.go

```diff
@@ -3,15 +3,84 @@ package raftstore
 import (
 	"encoding/binary"
 	"fmt"
+	"github.com/ngaut/log"
 	"github.com/pingcap/kvproto/pkg/eraftpb"
 	"github.com/pingcap/kvproto/pkg/metapb"
+	"github.com/pingcap/kvproto/pkg/pdpb"
 	"github.com/pingcap/kvproto/pkg/raft_cmdpb"
 	rspb "github.com/pingcap/kvproto/pkg/raft_serverpb"
 	"github.com/zhangjinpeng1987/raft"
 	"math"
 	"time"
 )
 
+type RegionChangeEvent int
+
+const (
+	RegionChangeEvent_Create RegionChangeEvent = 0 + iota
+	RegionChangeEvent_Update
+)
+
+type CoprocessorHost struct {
+	// Todo: currently it is a place holder
+}
+
+func (c *CoprocessorHost) PrePropose(region *metapb.Region, req *raft_cmdpb.RaftCmdRequest) error {
+	// Todo: currently it is a place holder
+	return nil
+}
+
+func (c *CoprocessorHost) OnRegionChanged(region *metapb.Region, event RegionChangeEvent, role raft.StateType) {
+	// Todo: currently it is a place holder
+}
+
+func (c *CoprocessorHost) OnRoleChanged(region *metapb.Region, role raft.StateType) {
+	// Todo: currently it is a place holder
+}
+
+type ApplyRouter struct {
+	// Todo: currently it is a place holder
+}
+
+func (a *ApplyRouter) ScheduleTask(regionId uint64, task *ApplyTask) {
+	// Todo: currently it is a place holder
+}
+
+type ReadyICPair struct {
+	Ready raft.Ready
+	IC    *InvokeContext
+}
+
+type PollContext struct {
+	Cfg             *Config
+	CoprocessorHost *CoprocessorHost
+	engine          *Engines
```

Comment:
Can we use `dbBoundle` instead of `engine` and remove this field?

Reply:
Because raft db will be used, like L475.

Reply:
Ok, I thought DBBundle contains raft db.